> Thanks @yusefkarim, this feedback was useful. I pushed follow-up commit [42af719](https://github.com/enjoy-digital/litex/commit/42af719466dd1b1df803c9621f1196e741c5bfff) to address the two issues you hit:
> 

Awesome, that resolved the issues yep. However, one more issue. To test you changes I decided to rebuild my OSS CAD based container image, and after that I ran into another issue that I think may due to the newer version of Yosys/yosys-slang:

```
2. Executing SLANG frontend.
Top level design units:
    ibex_top

.../pythondata_cpu_ibex/system_verilog/rtl/ibex_top.sv:550:36: error: reading net state during design initialization unsupported
    logic unused_scramble_inputs = scramble_key_valid_i & (|scramble_key_i) & (|RndCnstIbexKey) &
                                   ^~~~~~~~~~~~~~~~~~~~
.../ibex_top.sv:550:61: error: reading net state during design initialization unsupported
    logic unused_scramble_inputs = scramble_key_valid_i & (|scramble_key_i) & (|RndCnstIbexKey) &
                                                            ^~~~~~~~~~~~~~
.../ibex_top.sv:551:38: error: reading net state during design initialization unsupported
                                   (|scramble_nonce_i) & (|RndCnstIbexNonce) & scramble_req_q &
                                     ^~~~~~~~~~~~~~~~
Build failed: 3 errors, 0 warnings
ERROR: Compilation failed
```
After some digging I found this `slang` option: `--infer-input-ports-as-vars` [[ref](https://github.com/MikePopoloski/slang/blob/master/docs/command-line-ref.dox#L675)]. Adding that to `yosys_slang_opts` resolved the issue.

--- a/litex/soc/cores/cpu/ibex/core.py
+++ b/litex/soc/cores/cpu/ibex/core.py
@@ -297,9 +297,15 @@ class Ibex(CPU):
         if regfile == "fpga":
             platform.add_source(os.path.join(rtl_base, "ibex_register_file_fpga.sv"))
 
+        yosys_slang_opts = [
+            "--relax-enum-conversions",
+            "--ignore-initial",
+            "--infer-input-ports-as-vars",
+            "--top ibex_top",
+            "-G RegFile={}".format(regfiles[regfile]),
+        ]
         platform.yosys_use_slang  = True
-        platform.yosys_slang_opts = "--relax-enum-conversions --ignore-initial --top ibex_top -G RegFile={}".format(
-            regfiles[regfile])
+        platform.yosys_slang_opts = " ".join(yosys_slang_opts)
 
     def set_reset_address(self, reset_address):
         self.reset_address = reset_address

--- a/test/cores/test_cpu_ibex.py
+++ b/test/cores/test_cpu_ibex.py
@@ -44,6 +44,7 @@ class TestIbexSources(unittest.TestCase):
         self.assertTrue(platform.yosys_use_slang)
         self.assertIn("--top ibex_top", platform.yosys_slang_opts)
         self.assertIn("-G RegFile=0", platform.yosys_slang_opts)
+        self.assertIn("--infer-input-ports-as-vars", platform.yosys_slang_opts)
         self.assertNotIn("--ignore-unknown-modules", platform.yosys_slang_opts)
         self.assertEqual(source_languages[os.path.join("syn", "rtl", "prim_clock_gating.v")], "systemverilog")
         self.assertIn(os.path.join("vendor", "lowrisc_ip", "ip", "prim_generic", "rtl", "prim_generic_flop.sv"), sources)

